contracts/raidBasic.sol

```diff
@@ -102,7 +102,8 @@ contract RaidBasic is Initializable, Raid {
     function removeRewardWeapon(uint256 index) public restricted {
         require(index < weaponDrops.length, "Index out of bounds");
 
-        delete weaponDrops[index];
+        weaponDrops[index] = weaponDrops[weaponDrops.length - 1];
+        delete weaponDrops[weaponDrops.length - 1];
```

Comment:
The logic from before this PR and after this PR are equivalent so as far as the PVE is concerned, this PR is good, but it's actually probably not quite the logic what we wanted in the first place. The length of the array is left unchanged and there's just a zeroed-out spot in the array instead.

We should call `weaponDrops.pop()` after copying the last element - see suggestion. Popping the last element implicitly calls `delete` on the last element as well.

```suggestion
        weaponDrops[index] = weaponDrops[weaponDrops.length - 1];
        weaponDrops.pop();
```

Reply:
Implemented your suggestion, thanks.